core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -1504,22 +1504,37 @@ class UnifiedLog(@volatile var logStartOffset: Long,
 
   /**
    * Flush all local log segments
+   *
+   * @param inclusive should be true during a clean shutdown, and false otherwise. The reason is that
+   * we have to pass logEngOffset + 1 to the `localLog.flush(offset: Long): Unit` function to flush empty
+   * active segments, which is important to make sure we don't lose the empty index file during shutdown.
    */
-  def flush(): Unit = flush(logEndOffset)
+  def flush(inclusive: Boolean): Unit = flush(logEndOffset, inclusive)
 
   /**
    * Flush local log segments for all offsets up to offset-1
    *
    * @param offset The offset to flush up to (non-inclusive); the new recovery point
    */
-  def flush(offset: Long): Unit = {
-    maybeHandleIOException(s"Error while flushing log for $topicPartition in dir ${dir.getParent} with offset $offset") {
-      if (offset > localLog.recoveryPoint) {
-        debug(s"Flushing log up to offset $offset, last flushed: $lastFlushTime,  current time: ${time.milliseconds()}, " +
+  def flush(offset: Long): Unit = flush(offset, false)
```

Comment:
There is another flush method defined above with only one parameter, which is a boolean. It is easy to confuse that with this method, which also has only one parameter. So just for readability, does it make sense to call this as `def flushUptoOffsetExclusive(offset: Long)`?